lib/services/credentials/credential.service.js

```diff
@@ -45,84 +45,94 @@ s.insertCredential = function (id, type, credentialDetails) {
     throw new Error(`Invalid credential type: ${type}`); // TODO: replace with validation error
   }
 
-  return this.getCredential(id, type) // check if credential already exists
-    .then((cred) => {
-      if (cred && cred.isActive) {
-        throw new Error('Credential already exists and is active'); // TODO: replace with validation error
-      }
-      return dereferencePromise;
-    })
-    .then((resolvedSchema) => {
-      const credentialConfig = resolvedSchema.properties[type];
-      const newCredential = { isActive: 'true' };
-      utils.appendCreatedAt(newCredential);
-      utils.appendUpdatedAt(newCredential);
+  // TODO: not a good approach, new way TBD
+  const areMultipleCredsAllowed = ['key-auth', 'jwt'].includes(type);
+
+  let credentialExistsPromise;
+  if (areMultipleCredsAllowed) {
+    credentialExistsPromise = Promise.resolve(dereferencePromise);
```

Comment:
If this is a Promise there's no need to wrap it again
```suggestion
    credentialExistsPromise = dereferencePromise;
```